### PR TITLE
bpf: option for selecting DSR L4 DNAT method for IPIP

### PR DESCRIPTION
**Pull request by borkmann**
See commit msg

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

File: bpf/lib/lb.h [hunk 1]
```diff
@@ -185,7 +185,11 @@ struct bpf_elf_map __section_maps LB_AFFINITY_MATCH_MAP = {
 };
 #endif
 
-#define REV_NAT_F_TUPLE_SADDR 1
+#define REV_NAT_F_TUPLE_SADDR	1
+#ifndef DSR_XLATE_MODE
+# define DSR_XLATE_MODE		0
```
Comment (brb) [inline bpf/lib/lb.h:190]: Nit: is spacing before `0` is off or GH is not able to render it properly?

**Review by brb — APPROVED** [review]

File: bpf/lib/lb.h (hunk 1 above)
Comment (borkmann) [reply bpf/lib/lb.h:190]: It's the rendering: https://github.com/cilium/cilium/blob/pr/xlate-l4/bpf/lib/lb.h#L187

**Comment by borkmann** [comment]
retest-gke

**Comment by borkmann** [comment]
retest-1.16-netnext

**Comment by borkmann** [comment]
retest-1.20-4.19

**Comment by borkmann** [comment]
retest-1.21-4.9

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
retest-runtime

**Comment by borkmann** [comment]
retest-1.16-netnext

**Pull request merged**